test/python/WMCore_t/WMSpec_t/StdSpecs_t/StdBase_t.py

```diff
@@ -42,12 +43,39 @@ def testStdBaseValidation(self):
         i.e. no exception should be raised.
         """
         arguments = StdBase.getTestArguments()
-        print
         for k in sorted(arguments.keys()):
             print k, arguments[k]
         stdBaseInstance = StdBase()
         stdBaseInstance.factoryWorkloadConstruction("TestWorkload", arguments)
         return
+    
+    @attr('integration')
+    def testStdBaseIncludeParentsValidation(self):
+        """
+        _testStdBaseValidation_
+
+        Check that the test arguments pass basic validation,
+        i.e. no exception should be raised.
+        """
+        arguments = StdBase.getTestArguments()
+        stdBaseInstance = StdBase()
+        
+        arguments["IncludeParents"] = True
+        arguments["InputDataset"] = "/Cosmics/Commissioning2015-v1/RAW"
+        self.assertRaises(WMSpecFactoryException, stdBaseInstance.factoryWorkloadConstruction, "TestWorkload", arguments)
+        
+        arguments["IncludeParents"] = True
+        self.assertRaises(WMSpecFactoryException, stdBaseInstance.factoryWorkloadConstruction, "TestWorkload", arguments)
+        
+        arguments["IncludeParents"] = True
+        arguments["InputDataset"] = "/Cosmics/Commissioning2015-6Mar2015-v1/RECO"
+        stdBaseInstance.factoryWorkloadConstruction("TestWorkload", arguments)
+        
+        arguments["IncludeParents"] = False
+        arguments["InputDataset"] = "/Cosmics/Commissioning2015-6Mar2015-v1/RECO"
+        stdBaseInstance.factoryWorkloadConstruction("TestWorkload", arguments)
+        return
+        return
```

Comment:
Yes, sometimes, it doesn’t listen when I just tell that one time. :-)

On Mar 12, 2015, at 9:55 AM, Eric Vaandering <notifications@github.com<mailto:notifications@github.com>> wrote:


In test/python/WMCore_t/WMSpec_t/StdSpecs_t/StdBase_t.py<https://github.com/dmwm/WMCore/pull/5676#discussion_r26308291>:

> +        arguments["IncludeParents"] = True
> +        arguments["InputDataset"] = "/Cosmics/Commissioning2015-v1/RAW"
> +        self.assertRaises(WMSpecFactoryException, stdBaseInstance.factoryWorkloadConstruction, "TestWorkload", arguments)
> +
> +        arguments["IncludeParents"] = True
> +        self.assertRaises(WMSpecFactoryException, stdBaseInstance.factoryWorkloadConstruction, "TestWorkload", arguments)
> +
> +        arguments["IncludeParents"] = True
> +        arguments["InputDataset"] = "/Cosmics/Commissioning2015-6Mar2015-v1/RECO"
> +        stdBaseInstance.factoryWorkloadConstruction("TestWorkload", arguments)
> +
> +        arguments["IncludeParents"] = False
> +        arguments["InputDataset"] = "/Cosmics/Commissioning2015-6Mar2015-v1/RECO"
> +        stdBaseInstance.factoryWorkloadConstruction("TestWorkload", arguments)
> +        return
> +        return


Best to make sure....

—
Reply to this email directly or view it on GitHub<https://github.com/dmwm/WMCore/pull/5676/files#r26308291>.